src/pages/index.js

```diff
@@ -2,76 +2,215 @@ import "./index.css";
+function renderCard(card) {
+  const cardElement = createCard(card);
+  itemSection.addItem(cardElement);
 }
 
-function createCard(cardData) {
-  return new Card(cardData, cardTemplateSelector, handleImageClick).getView();
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    // a = true
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+const deleteConfirm = new PopupWithConfirmation("#card-delete-modal");
+const handleDeleteConfirm = (card) =>
+  api
+    .deleteCard(card._id)
+    .then(() => {
+      console.log(card);
+      card.remove();
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+deleteConfirm.setEventListeners();
+
+function handleDeleteClick(card) {
+  // Set the submit action for the confirmation popup
+  console.log(card._id);
+  // why don't we open confirm modal first?
+  deleteConfirm.setSubmitAction(() => {
+    console.log(deleteConfirm);
+    handleDeleteConfirm(card);
+    deleteConfirm.close();
+  });
 }
 
-function renderCard(cardData) {
-  const cardElement = createCard(cardData);
-  itemSection.addItem(cardElement);
+function handleLikeClick(card) {
+  api
+    .toggleCardLike(card._id, card.isLiked)
+    .then((_isLiked) => {
+      // console.log("RESPONSE");
+      // console.log(_isLiked);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+}
+
+function handleChangePicture(avatar) {
+  changePictureModal.setEventListeners();
+  console.log(avatar);
+  this._buttonElement = this._popupForm.querySelector(".modal__button");
+  this._buttonElement;
+  changePictureModal.renderLoading(true);
+  api
+    .setUserAvatar(avatar)
+    .then((data) => {
+      changePictureModal.renderLoading(false);
+      console.log(data.avatar);
+      userInfo.setUserAvatar(data.avatar);
+    })
+    .catch((err) => {
+      console.error(err);
+    });
+
+  changePictureModal.close();
 }
 
-function handleProfileEditSubmit(inputValues) {
-  userInfo.setUserInfo(inputValues.title, inputValues.description);
+const changePictureModal = new PopupWithForm(
+  "#profile-change-picture",
+  handleChangePicture
+);
+changePictureModal.setEventListeners();
+const pictureFormValidator = new FormValidator(config, pictureChangeForm);
+pictureFormValidator.enableValidation();
+profilePicture.addEventListener("click", () => {
+  console.log("not working");
+  pictureFormValidator. resetValidation();
+  changePictureModal.open();
+});
+
+const editProfileModal = new PopupWithForm(
+  "#profile-edit-modal",
+  (userData) => {
+    editProfileModal.renderLoading(true);
+    console.log(userData);
+    api
+      .saveUserData(
+        { name: userData.title, about: userData.description },
+        userData.avatar
+      )
+      .then((res) => {
+        handleProfileEditSubmit(res);
+        editProfileModal.renderLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+    function handleProfileEditSubmit(userData) {
```

Comment:
`handleProfileEditSubmit`

It's not a submit handler


You set user info here


`setUserInfo`